server/routes/fishApiRoutes.js

```diff
@@ -114,7 +114,25 @@ module.exports = function(app) {
         }
         // TODO make this promise based - or callback based
         res.sendStatus(204); // we're all good
+    });
+
+    app.put('/api/stopSellingThisFish/', function(req, res){
+        let thisId = req.body.target;
 
+        db.UserFish.findOne({
+            where: {
+                id: thisId
+            }    
+        })
+        .then((fish) => {
+            fish.update({forSale: 0})
+            .then (() => {
+                res.sendStatus(204);
+            })
+            .catch((error) => {
+                res.sendStatus(400).json(error);
+            })
```

Comment:
Could flatten this to 
```js
.then((fish) => fish.update({forSale: o}))
.then(() => res.sendStatus(204))
.catch((error) => res.sendStatus(400).json(error));
```

Reply:
Returning a promise within a promise will not hit the next `.then` until it completes :)